Add access-extension-notification scheduled task sample for OIG

mgere-oracle opened the pull request:
Will follow up with a complementary web service asset shortly.

oheimburger commented on security/identity-and-access-management/oracle-identity-governance/scheduled-task-samples/access-extension-notification/files/src/com/oracle/sample/AccessExtensionNotification.java:
```diff
@@ -0,0 +1,526 @@
+		logger.log(Level.INFO, "The scheduled task run has finished.");
+	}
+
+	private Boolean processAccounts(String userKey, List<Account> userApplications) throws SQLException {
```
Can the parameters be empty?

mgere-oracle replied:
Not with the current implementation, however I've added validations in case someone decides to further customize the code.

oheimburger commented on security/identity-and-access-management/oracle-identity-governance/scheduled-task-samples/access-extension-notification/files/src/com/oracle/sample/AccessExtensionNotification.java:
```diff
@@ -0,0 +1,526 @@
+		return (sendEmailAccounts || sendEmailEntitlements);
+	}
+
+	private Boolean processEntitlements(String userKey, Account currentAccount) throws SQLException {
```
Can the parameters be empty?

mgere-oracle replied:
Same as above.

oheimburger commented on security/identity-and-access-management/oracle-identity-governance/scheduled-task-samples/access-extension-notification/files/src/com/oracle/sample/AccessExtensionNotification.java:
```diff
@@ -0,0 +1,526 @@
+	 * @throws SQLException
+	 * @throws Exception
+	 */
+	private HashMap<String, String> dynamicTemplateFill(HashMap<String, String> attrMap, String templateName)
```
Can these parameters be empty?

mgere-oracle replied:
Same as above.

oheimburger commented on security/identity-and-access-management/oracle-identity-governance/scheduled-task-samples/access-extension-notification/files/src/com/oracle/sample/AccessExtensionNotification.java:
```diff
@@ -0,0 +1,526 @@
+			logger.log(Level.WARNING, "Exception encountered: ", e);
+			throw new SchedulerException("Email template fill has failed.");
+		} finally {
+			if (templateRS != null && !templateRS.isClosed())
```
Use {} even for single line blocks.

mgere-oracle replied:
Fixed.

oheimburger commented on security/identity-and-access-management/oracle-identity-governance/scheduled-task-samples/access-extension-notification/files/src/com/oracle/sample/AccessExtensionNotification.java:
```diff
@@ -0,0 +1,526 @@
+			} catch (Exception e) {
+				logger.log(Level.WARNING, "Entitlement processing failed with: ", e);
+			} finally {
+				if (resultSet != null && !resultSet.isClosed())
```
Use {} even for single line blocks.

mgere-oracle replied:
Fixed.

oheimburger commented on security/identity-and-access-management/oracle-identity-governance/scheduled-task-samples/access-extension-notification/files/src/com/oracle/sample/AccessExtensionNotification.java:
```diff
@@ -0,0 +1,526 @@
+			} catch (Exception e) {
+				logger.log(Level.WARNING, "Account processing failed with: ", e);
+			} finally {
+				if (resultSet != null && !resultSet.isClosed())
```
Use {} even for single line blocks.

mgere-oracle replied:
Fixed.

AlexanderHodicke commented:
@mgere-oracle Please see olafs review and fix them. Otherwise please close the PR.

mgere-oracle commented:
> @mgere-oracle Please see olafs review and fix them. Otherwise please close the PR.

I have addressed all the points raised by @oheimburger. Please let me know if there's anything left pending from my side. I would like to have this merged please, not closed.
